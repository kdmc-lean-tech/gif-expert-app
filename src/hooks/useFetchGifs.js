import { useEffect, useState } from 'react';
import { fetchGifs } from '../helpers/getGifs';

export const useFetchGifs = (category) => {
  const [state, setState] = useState({ loading: true, results: [] });

  useEffect(() => {
    fetchGifs(category)
      .then(gifs => setState({
        loading: false,
        results: gifs
      }));
  }, [category]);

  return {...state};
}
