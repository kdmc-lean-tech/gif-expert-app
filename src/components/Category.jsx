import React from 'react';
import GifItem from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { Container, Row, Col } from 'react-bootstrap';
import PropTypes from 'prop-types';

const Category = ({ category }) => {

  const { loading, results } = useFetchGifs(category);

  return (
    <Container className='mb-4'>
      <Row>
        {
          loading && <p>Loading........</p>
        }
        {
          results.map(gif =>
            <Col lg='3' md='6' key={ gif.id } className='mb-4'>
              <GifItem gif={ gif } key={ gif.id } />
            </Col>
          )
        }
      </Row>
    </Container>
  );
}

Category.propTypes = {
  category: PropTypes.string.isRequired
}

export default Category;
