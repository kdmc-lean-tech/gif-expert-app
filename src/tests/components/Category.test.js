import { shallow } from 'enzyme';
import Category from '../../components/Category';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Category Component Test', () => {
  test('should render category component', () => {

    useFetchGifs.mockReturnValue({
      results: [],
      loading: true
    });

    const category = 'gohan';
    const wrapper = shallow(<Category category={ category } />);
    expect(wrapper).toMatchSnapshot();
  });
  test('should show items when be upload gifs useFecthGifs', () => {
    const results = [
      {
        url: 'https://media0.giphy.com/media/1dMNqVx9Kb12EBjFrc/giphy.gif?cid=a91e1fa02z4huvfdz4isenmiuf5oyayjiodcifdstfjnk4ci&amp;rid=giphy.gif"',
        title: 'Bugs Bonny',
        id: '3'
      }
    ];
    useFetchGifs.mockReturnValue({
      results,
      loading: false
    });
    const category = 'gohan';
    const wrapper = shallow(<Category category={ category } />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifItem').length).toBe(results.length);
  });
});
