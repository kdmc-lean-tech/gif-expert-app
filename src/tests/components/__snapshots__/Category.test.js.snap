// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category Component Test should render category component 1`] = `
<Container
  className="mb-4"
  fluid={false}
>
  <Row
    noGutters={false}
  >
    <p>
      Loading........
    </p>
  </Row>
</Container>
`;

exports[`Category Component Test should show items when be upload gifs useFecthGifs 1`] = `
<Container
  className="mb-4"
  fluid={false}
>
  <Row
    noGutters={false}
  >
    <Col
      className="mb-4"
      key="3"
      lg="3"
      md="6"
    >
      <GifItem
        gif={
          Object {
            "id": "3",
            "title": "Bugs Bonny",
            "url": "https://media0.giphy.com/media/1dMNqVx9Kb12EBjFrc/giphy.gif?cid=a91e1fa02z4huvfdz4isenmiuf5oyayjiodcifdstfjnk4ci&amp;rid=giphy.gif\\"",
          }
        }
        key="3"
      />
    </Col>
  </Row>
</Container>
`;
